### PR TITLE
Feature/ally

### PR DESCRIPTION
**Pull request by yalania**
- Dialogues for ally
- Lightball recieve from ally
- Move around the scene

Note: Right now you can skip recieving the lightball, I will continue developing it once we have this merge.

### CODE REVIEW

File: Scripts/src/Script/AllyInteractionsScript.cpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	{
+		can_start_next_dialogue = false;
+		camera->active = false;
+	}
+}
+
+//Copied from mushdoom
+void AllyInteractionsScript::RotateTowardsPlayer() const
+{
+	float3 desired_direction = player_1->player_go->transform.GetGlobalTranslation() - owner->parent->transform.GetGlobalTranslation();
```
Comment (Dartemiss) [inline Scripts/src/Script/AllyInteractionsScript.cpp:165]: It only looks to the player1? Maybe could Look into the closer player instead

Comment (yalania) [reply Scripts/src/Script/AllyInteractionsScript.cpp:175]: in most multi games only the player one can trigger conversations etc, so I just used player1

File: Scripts/src/Script/LightBall.cpp [hunk 2]
```diff
@@ -102,7 +93,7 @@ void LightBall::Update()
 		return;
 	}
 
-	if(input_manager->GetGameInput("LightBall", static_cast<PlayerID>(current_player_controller->player - 1)))
+	if(!mAllyHasLightBall && input_manager->GetGameInput("LightBall", static_cast<PlayerID>(current_player_controller->player - 1)))
```
Comment (JorxSS) [inline Scripts/src/Script/LightBall.cpp:96]: cs here

File: Scripts/src/Script/LightBall.h [hunk 3]
```diff
@@ -158,6 +159,9 @@ class LightBall : public Script
 
 	//Delete when the animations and STM work better
 	float frame_counter = 0.f;
+
+	//Ally LightBall Tutorial
+	bool mAllyHasLightBall = true;
```
Comment (JorxSS) [inline Scripts/src/Script/LightBall.h:164]: cs here with this variable

File: Scripts/src/Script/Script.cpp [hunk 4]
```diff
@@ -91,5 +91,6 @@ void Script::ShowDraggedObjects()
 
 Script::~Script()
 {
+	//delete panel;
```
Comment (JorxSS) [inline Scripts/src/Script/Script.cpp:94]: ???

Comment (yalania) [reply Scripts/src/Script/Script.cpp:94]: I have seen there are a lot of memory leaks in the gameplay project, this an example but need to add ENGINE_API to the destructor, as you are going to test it with master branch I left it commented so you don't have linking issues

Comment (Dartemiss) [reply Scripts/src/Script/Script.cpp:94]: Which memory leaks did you find?

Comment (yalania) [reply Scripts/src/Script/Script.cpp:94]: just do a search for = new and you will se most of the allocated stuff is never deleted, right now I don't remember all of them

Comment (Dartemiss) [reply Scripts/src/Script/Script.cpp:94]: wow u are right, all the instances and the states of the STM of enemies.

**Review by JorxSS — APPROVED** [review]
Sorry for the delay
image

**Review by Dartemiss — APPROVED** [review]
GJ Anabel! :D

**Pull request merged**